File: confirm_the_ending/confirm_the_ending.py
'''
Check if a string (first argument, string) ends with the given target string (second argument, target).


>>> confirm_ending("Bastian", "n")
True

>>> confirm_ending("Open sesame", "pen")
False

>>> confirm_ending("Connor", "n")
False

>>> confirm_ending("If you want to save our world, you must hurry. We dont know how much longer we can withstand the nothing", "mountain")
False

>>> confirm_ending("Walking on water and developing software from a specification are easy if both are frozen", "specification")
False

>>> confirm_ending("He has to give me a new name", "name")
True

>>> confirm_ending("Open sesame", "same")
True

>>> confirm_ending("Open sesame", "pen")
False

'''

# example 1
def confirm_ending(string: str, target: str) -> bool:
    if string[-len(target):] == target:
        return True
    else:
        return False

# example 2
def confirm_ending(string: str, target: str) -> bool:
    if string.endswith(target):
        return True
    else:
        return False

# example 3
def confirm_ending(string: str, target: str) -> bool:
    # Start slice from last char (-1), stepping backward one char at a time (-1), the length of target -(len(target)+1)
    slice_at = slice(-1, -(len(target)+1), -1)
    # slice_at is ordered backwards so reverse to make forward order
    str_end_forward = string[slice_at][::-1]
    # Compare forward string with target
    if str_end_forward  == target:
        return True
    else:
        return False

File: confirm_the_ending/confirm_the_ending.js

// JavaScript 5
// Use .substring(indexStart, indexEndToExclude)
function confirmEnding(str, target) {
  "use strict";
  if (str.substring(str.length - target.length) !== target) {
    return false;
  } else {
      console.log(true);
      return true;
  }
}

// Use .slice(negIndex)
function confirmEnding2(str, target) {
  "use strict";
  // Negative index integer == numb of chars to include from end
  var strEnding = str.slice(-target.length);
  if (strEnding !== target) {
    return false;
  }
  console.log(true);
  return true;
}

// Use .substr(start, chars to extract)
function confirmEnding3(str, target) {
  "use strict";
  if (str.substr(str.length-1, target.length) !== target) {
    return false;
  } else {
      console.log(true);
      return true;
  }
}

// Use .endsWith(searchString)
function confirmEnding4(str, target) {
  "use strict";
  if (str.endsWith(target)) {
      console.log(true);
      return true;
  } else {
    return false;
  }
}

// Use .includes(searchString, startPos)
function confirmEnding5(str, target) {
  "use strict";
  // Start at X number of chars from str's last letter
  var startPos = (str.length-1) - target.length;

  if (str.includes(target, startPos)) {
      console.log(true);
      return true;
  } else {
    return false;
  }
}


// JavaScript 6
const confirmEnding6 = (str, target) => str.endsWith(target) ? true : false


confirmEnding("Bastian", "n");
confirmEnding("Open sesame", "pen");
confirmEnding2("Connor", "n");
confirmEnding2("If you want to save our world, you must hurry. We dont know how much longer we can withstand the nothing", "mountain");
confirmEnding3("Walking on water and developing software from a specification are easy if both are frozen", "specification");
confirmEnding4("He has to give me a new name", "name");
confirmEnding5("Open sesame", "same");
confirmEnding6("Walking on water and developing software from a specification are easy if both are frozen", "specification");
confirmEnding6("He has to give me a new name", "name");


// JavaScript 6
